src/app/rpmostree-libbuiltin.c

```diff
@@ -42,21 +42,47 @@ rpmostree_usage_error (GOptionContext  *context,
   (void) glnx_throw (error, "usage error: %s", message);
 }
 
+static char*
+get_id_from_deployment_variant (GVariant *deployment)
+{
+  g_autoptr(GVariantDict) dict = g_variant_dict_new (deployment);
+  g_autofree char *id;
+  g_assert (g_variant_dict_lookup (dict, "id", "s", &id));
+  return g_steal_pointer (&id);
```

Comment:
Why not just `g_variant_lookup(deployment, "id", "&s", &id)` ?

Reply:
Yeah, I had it return a new string because we weren't keeping a reference to the associated `GVariant`. In the new approach, we do.